### PR TITLE
Kit's Gunpowder Weapons - CE Patch

### PR DESCRIPTION
**Pull request by Calvinhood1**
## Additions

CE Patch for Kit's Gunpowder Weapons.  (May need balancing)

Guns are compatible.

Turrets are compatible.

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (5-10 minutes total of testing each individual gun + ammo excluding cannon balls)

### CODE REVIEW

**Comment by Calvinhood1** [comment]
I'll be honest. This is my first time I've ever tried to patch turrets, use github, and some other things. Feel free to close the request if I messed something up too drastically.

**Comment by randomtyper** [comment]
Use the "Not just a cannon" mod patch as a reference. Sorry, but I can't help you otherwise right now.

File: Patches/Kit's Gunpowder Weapons/Defs/Ammo/AmmoCategoryDefs/AmmoCategories_Medieval.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Defs>
+
+	<CombatExtended.AmmoCategoryDef>
```
Comment (randomtyper) [inline Patches/Kit's Gunpowder Weapons/Defs/Ammo/AmmoCategoryDefs/AmmoCategories_Medieval.xml:4]: There already are musket balls in the Combat Extended, no need to create a duplicate.

Comment (Calvinhood1) [reply Patches/Kit's Gunpowder Weapons/Defs/Ammo/AmmoCategoryDefs/AmmoCategories_Medieval.xml:4]: My bad, replaced it with CE's.

**Comment by Calvinhood1** [comment]
The "Not Only just a cannon" ("Nothing but just a cannon") mod patch seems to be outdated and does odd things that none of the other patches seem to do. I attempted to fix the turret patch by cross referencing with turret patches from "Reich Armory," "DefensiveMachineGunTurretPack," and "More Vanilla Turrets," but didn't really end up anywhere. I am honestly stumped on patching these turrets.

**Comment by randomtyper** [comment]
Yeah, I tested it recently. Weird stuff happens. I'll create an issue later. And that patch is recent, mind you. Maybe try Turret Collection patch? It is an old patch that worked good enough.

**Comment by Calvinhood1** [comment]
Everything is patched and functional other than a part of the Bombard. I haven't figured out how to get it to work as a mortar, so when a pawn tries to shoot it, the projectile spawns on the tile of the Bombard and explodes on said tile. 

Thank you for the suggested patches for reference. I apologize for calling the "Nothing but just a cannon" outdated, but the patch I saw was made 2 years ago and only for Rimworld version 1.0, so I assumed otherwise.

**Comment by Calvinhood1** [comment]
Patch finished, probably needs balancing. Decided to cut out ##Alternatives, will just make a separate ammo patch for that.

**Comment by N7Huntsman** [comment]
Thank you for putting the work in to get it all laid out. All the pieces and parts are here, but the formatting is a bit all over the place, so I ended up just transferring the contents over to some reformatted files and opening a separate PR. Patch is now merged.

**Pull request closed without merging**